system_parameters_tests.py

```diff
@@ -0,0 +1,1042 @@
+import unittest
+import utils
+import config
+import requests
+import json
+import funcs
+import os
+import time
+
+
+class SystemParametersCase(unittest.TestCase):
+    def setUp(self):
+        global url, token, prKey, pause,\
+            dbHost, dbName, login, password, contractName, maxInt, minInt, defaultValues, defaultValueKey
+        self.config = config.getNodeConfig()
+        url = self.config["2"]["url"]
+        pause = self.config["1"]["time_wait_tx_in_block"]
+        prKey = self.config["1"]['private_key']
+        self.data = utils.login(url, prKey)
+        token = self.data["jwtToken"]
+        dbHost = self.config["2"]["dbHost"]
+        dbName = self.config["2"]["dbName"]
+        login = self.config["2"]["login"]
+        password = self.config["2"]["pass"]
+        contractName = "UpdateSysParam"
+        maxInt = 9223372036854775807
+        minInt = -9223372036854775808
+        defaultValueKey = "system_parameters"
+        defaultValues = self.readDefaultParameters()
+
+    def readDefaultParameters(self):
+        path = os.path.join(os.getcwd(), "defaultSystemParameters.json")
+        with open(path, 'r') as f:
+            data = f.read()
+        return json.loads(data)
+
+    def assertTxInBlock(self, result, jwtToken):
+        self.assertIn("hash", result)
+        hash = result['hash']
+        status = utils.txstatus(url, pause, hash, jwtToken)
+        if len(status['blockid']) > 0:
+            self.assertNotIn(json.dumps(status), 'errmsg')
+            return status["blockid"]
+        else:
+            return status["errmsg"]["error"]
+
+    def check_get_api(self, endPoint, data, keys):
+        end = url + endPoint
+        result = funcs.call_get_api(end, data, token)
+        for key in keys:
+            self.assertIn(key, result)
+        return result
+
+    def call(self, name, data):
+        resp = utils.call_contract(url, prKey, name, data, token)
+        resp = self.assertTxInBlock(resp, token)
+        return resp
+
+    def create_contract(self, url, prKey):
+        code,name = utils.generate_name_and_code("")
+        data = {'Wallet': '', 'Value': code, "ApplicationId": 1,
+                'Conditions': "ContractConditions(`MainCondition`)"}
+        resp = utils.call_contract(url, prKey, "NewContract", data, self.data1["jwtToken"])
+        return name
+
+    def getCountBlocks(self):
+        return utils.getCountTable(dbHost, dbName, login, password, "block_chain")
+
+    def getSystemParameterValue(self, name):
+        return utils.getSystemParameterValue(dbHost, dbName, login, password, name)
+
+
+
+    #default_ecosystem_page
+    #default_ecosystem_menu
+    #default_ecosystem_contract
+
+    def test_gap_between_blocks(self):
+        name = "gap_between_blocks"
+        data1 = {"Name": name, "Value": "-1"}
```

Comment:
why not just use cycle
`
invalidValues = ["-1", "0", "86400", "hello", "1452.78", "3"]
for value in invalidValues:
      self.call(contractName, {"Name": name, "Value": value})
`

Reply:
The same about all tests in that file